corehq/apps/hqwebapp/forms.py

```diff
@@ -48,22 +51,35 @@ def clean(self):
         if settings.ADD_CAPTCHA_FIELD_TO_FORMS:
             if not self.cleaned_data.get('captcha'):
                 raise ValidationError(_("Please enter valid CAPTCHA"))
-
-        try:
-            cleaned_data = super(EmailAuthenticationForm, self).clean()
-        except ValidationError:
-            user = CouchUser.get_by_username(username)
-            if user and user.is_locked_out():
-                metrics_counter('commcare.auth.lockouts')
-                raise ValidationError(LOCKOUT_MESSAGE)
-            else:
-                raise
         user = CouchUser.get_by_username(username)
         if user and user.is_locked_out():
             metrics_counter('commcare.auth.lockouts')
             raise ValidationError(LOCKOUT_MESSAGE)
+
+        cleaned_data = super(EmailAuthenticationForm, self).clean()
+        if not user:
+            try:
+                ConsumerUser.objects.get(user=self.user_cache)
+                EmailAuthenticationForm.create_web_user_from_consumer_user(username,
+                                                                           password,
+                                                                           get_ip(self.request))
+            except ConsumerUser.DoesNotExist:
+                self.get_invalid_login_error()
```

Comment:
do you have to raise this somehow? I don't think this does anything right now.

Reply:
made the change